test/ag-sparql-federate-mocha.js

```diff
@@ -0,0 +1,66 @@
+// ag-sparql-federate-mocha.js
+
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.should();
+chai.use(chaiAsPromised);
+
+var _ = require('underscore');
+var http = require('http');
+var os = require('os');
+var fs = require('fs');
+var path = require('path');
+var noflo = require('noflo');
+var test = require('./common-test');
+
+describe('ag-sparql-federate', function() {
+    var port = 1337;
+    var origin = "http://localhost:" + port;
+    it("should create and close a session with authentication", function() {
+        this.timeout(5000);
+        var requests = 0;
+        var session;
+        return new Promise(function(done, fail) {
+            var server = http.createServer();
+            afterEach(_.once(server.close.bind(server)));
+            server.on('request', function(req, res) {
+                requests++;
```

Comment:
I thought the point of federating was to work across multiple servers within a cluster?  This is single server though?  This looks to me like it tests file access using the API, but does not actually test the distributed aspect.  Am I reading this right?  Should we have a test that verifies we actually get the data from a couple servers or is that not possible in a mocha test?   Have we tried this kind of thing on our 10 node cluster?